### PR TITLE
python3Packages.pytest-helpers-namespace: add missing dependency

### PR DESCRIPTION
**Pull request by austinbutler**
###### Motivation for this change

Found `ocrmypdf` was broken, I think by https://github.com/NixOS/nixpkgs/pull/117452 where `pytest-helpers-namespace` was upgraded but was missing a dependency.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by austinbutler** [comment]
Result of `nixpkgs-review pr 121122` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>img2pdf (python38Packages.img2pdf)</li>
    <li>ocrmypdf</li>
    <li>pdfarranger</li>
    <li>python38Packages.pikepdf</li>
    <li>python38Packages.pypiserver</li>
    <li>python38Packages.pytest-helpers-namespace</li>
    <li>python38Packages.setuptools-declarative-requirements</li>
    <li>python39Packages.img2pdf</li>
    <li>python39Packages.pikepdf</li>
    <li>python39Packages.pypiserver</li>
    <li>python39Packages.pytest-helpers-namespace</li>
    <li>python39Packages.setuptools-declarative-requirements</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121122` at 23aa18c2 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built successfully:</summary>
  <ul>
    <li>img2pdf (python38Packages.img2pdf)</li>
    <li>ocrmypdf</li>
    <li>pdfarranger</li>
    <li>python38Packages.pikepdf</li>
    <li>python38Packages.pypiserver</li>
    <li>python38Packages.pytest-helpers-namespace</li>
    <li>python38Packages.setuptools-declarative-requirements</li>
    <li>python39Packages.img2pdf</li>
    <li>python39Packages.pikepdf</li>
    <li>python39Packages.pypiserver</li>
    <li>python39Packages.pytest-helpers-namespace</li>
    <li>python39Packages.setuptools-declarative-requirements</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchPypi`.
Near pkgs/development/python-modules/pypiserver/default.nix:1:23:
```
  |
1 | { buildPythonPackage, fetchPypi, fetchFromGitHub, lib, passlib, pytestCheckHook
  |                       ^
```
</li>
    <li>warning: unused-argument

Unused argument: `setuptools`.
Near pkgs/development/python-modules/setuptools-declarative-requirements/default.nix:1:68:
```
  |
1 | { buildPythonPackage, fetchPypi, lib, pypiserver, pytestCheckHook, setuptools
  |                                                                    ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/setuptools-declarative-requirements/default.nix:23:0:
```
   |
23 |     description = "Declarative setuptools Config Requirements Files Support";
   | ^
```
</li>
    <li>warning: unused-argument

Unused argument: `setuptools_scm`.
Near pkgs/development/python-modules/pypiserver/default.nix:2:31:
```
  |
2 | , setuptools, setuptools-git, setuptools_scm, twine, webtest }:
  |                               ^
```
</li>
  </ul>
</details>

**Comment by austinbutler** [comment]
Result of `nixpkgs-review pr 121122` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>img2pdf (python38Packages.img2pdf)</li>
    <li>python38Packages.pikepdf</li>
    <li>python38Packages.pypiserver</li>
    <li>python38Packages.pytest-helpers-namespace</li>
    <li>python38Packages.setuptools-declarative-requirements</li>
    <li>python39Packages.img2pdf</li>
    <li>python39Packages.pikepdf</li>
    <li>python39Packages.pypiserver</li>
    <li>python39Packages.pytest-helpers-namespace</li>
    <li>python39Packages.setuptools-declarative-requirements</li>
  </ul>
</details>

File: pkgs/development/python-modules/setuptools-declarative-requirements/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  buildInputs = [ setuptools_scm ];
+
+  checkInputs = [ pypiserver pytestCheckHook virtualenv ];
+
+  # Tests use network
+  doCheck = false;
+
+  meta = with lib; {
+    homepage =
+      "https://github.com/s0undt3ch/setuptools-declarative-requirements";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/setuptools-declarative-requirements/default.nix:22]:
```suggestion
    homepage = "https://github.com/s0undt3ch/setuptools-declarative-requirements";
```

File: pkgs/development/python-modules/setuptools-declarative-requirements/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ buildPythonPackage, fetchPypi, lib, pypiserver, pytestCheckHook
+, setuptools_scm, virtualenv }:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/setuptools-declarative-requirements/default.nix:2]:
```suggestion
, setuptools-scm, virtualenv }:
```

File: pkgs/development/python-modules/setuptools-declarative-requirements/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    sha256 = "1l8zmcnp9h8sp8hsw7b81djaa1a9yig0y7i4phh5pihqz1gdn7yi";
+  };
+
+  buildInputs = [ setuptools_scm ];
+
+  checkInputs = [ pypiserver pytestCheckHook virtualenv ];
+
+  # Tests use network
+  doCheck = false;
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/setuptools-declarative-requirements/default.nix:19]:
```suggestion
  buildInputs = [ setuptools-scm ];

  checkInputs = [ pypiserver pytestCheckHook virtualenv ];

  # Tests use network
  doCheck = false;
  
  pythonImportsCheck = [ "declarative_requirements" ];

```

File: pkgs/development/python-modules/pytest-helpers-namespace/default.nix [hunk 4]
```diff
@@ -1,30 +1,26 @@
 { buildPythonPackage
-, fetchFromGitHub
-, pytest
+, fetchPypi
+, pytestCheckHook
+, isPy27
 , lib
+, setuptools
+, setuptools-declarative-requirements
+, setuptools_scm
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pytest-helpers-namespace/default.nix:8]:
```suggestion
, setuptools-scm
```

File: pkgs/development/python-modules/pytest-helpers-namespace/default.nix [hunk 5]
```diff
@@ -1,30 +1,26 @@
...
-    repo = pname;
-    rev = version;
-    sha256 = "0ikwiwp9ycgg7px78nxdkqvg7j97krb6vzqlb8fq8fvbwrj4q4v2";
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0pyj2d45zagmzlajzqdnkw5yz8k49pkihbydsqkzm413qnkzb38q";
   };
 
-  buildInputs = [ pytest ];
+  buildInputs = [ setuptools setuptools_scm setuptools-declarative-requirements ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pytest-helpers-namespace/default.nix:21]:
```suggestion
  nativeBuildInputs = [ setuptools setuptools-declarative-requirements setuptools-scm ];
```

File: pkgs/development/python-modules/pytest-helpers-namespace/default.nix [hunk 6]
```diff
@@ -1,30 +1,26 @@
...
 
-  checkInputs = [ pytest ];
-
-  checkPhase = ''
-    pytest
-  '';
-
-  # The tests fail with newest pytest. They passed with pytest_3, which no longer exists
-  doCheck = false;
+  checkInputs = [ pytestCheckHook ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pytest-helpers-namespace/default.nix:24]:
```suggestion

  pythonImportsCheck = [ "pytest_helpers_namespace" ];

```

File: pkgs/development/python-modules/pypiserver/default.nix [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+  preCheck = ''
+    export HOME=$TMPDIR
+  '';
+
+  checkInputs = [ passlib pytestCheckHook twine webtest ];
+
+  # These tests try to use the network
+  disabledTests =
+    [ "test_pipInstall_openOk" "test_pipInstall_authedOk" "test_hash_algos" ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pypiserver/default.nix:27]:
```suggestion
  disabledTests = [
    "test_pipInstall_openOk"
    "test_pipInstall_authedOk"
    "test_hash_algos"
  ];
```

Why does test_hash_algos fail?

File: pkgs/development/python-modules/pypiserver/default.nix [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pypiserver/default.nix:28]:
```suggestion

  pythonImportsCheck = [ "pypiserver" ];

```

File: pkgs/development/python-modules/pypiserver/default.nix [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  version = "1.4.2";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1z5rsmqgin98m6ihy1ww42fxxr6jb4hzldn8vlc9ssv7sawdz8vz";
+  };
+
+  buildInputs = [ setuptools-git ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pypiserver/default.nix:15]:
```suggestion
  natibeBuildInputs = [ setuptools-git ];
```

File: pkgs/development/python-modules/setuptools-declarative-requirements/default.nix [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    sha256 = "1l8zmcnp9h8sp8hsw7b81djaa1a9yig0y7i4phh5pihqz1gdn7yi";
+  };
+
+  buildInputs = [ setuptools-scm ];
+
+  checkInputs = [ pypiserver pytestCheckHook virtualenv ];
+
+  # Tests use network
+  doCheck = false;
+
```
Comment (mweinelt) [inline pkgs/development/python-modules/setuptools-declarative-requirements/default.nix:19]: trailing whitespaces here, the editor config check complains :)

File: pkgs/development/python-modules/pypiserver/default.nix [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+  version = "1.4.2";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1z5rsmqgin98m6ihy1ww42fxxr6jb4hzldn8vlc9ssv7sawdz8vz";
+  };
+
+  natibeBuildInputs = [ setuptools-git ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/pypiserver/default.nix:15]:
```suggestion
  nativeBuildInputs = [ setuptools-git ];
```

Typo, did you test this?

**Review by mweinelt — CHANGES_REQUESTED** [review]

**Comment by mweinelt** [comment]
Looks pretty good otherwise!

File: pkgs/development/python-modules/pypiserver/default.nix (hunk 11 above)
Comment (austinbutler) [reply pkgs/development/python-modules/pypiserver/default.nix:15]: Interestingly I did... I'll take another look and see if maybe it's not actually needed.

Comment (mweinelt) [reply pkgs/development/python-modules/pypiserver/default.nix:15]: It is needed!

Comment (austinbutler) [reply pkgs/development/python-modules/pypiserver/default.nix:15]: Hah, yep, confirmed! Will fix up. Sorry for the mess, have been trying to patch this up in between doing a bunch of other things.

File: pkgs/development/python-modules/pypiserver/default.nix (hunk 7 above)
Comment (mweinelt) [reply pkgs/development/python-modules/pypiserver/default.nix]: Also network related. It tries something with twine and what not.

**Comment by austinbutler** [comment]
Result of `nixpkgs-review pr 121122` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>img2pdf (python38Packages.img2pdf)</li>
    <li>ocrmypdf</li>
    <li>pdfarranger</li>
    <li>python38Packages.pikepdf</li>
    <li>python38Packages.pypiserver</li>
    <li>python38Packages.pytest-helpers-namespace</li>
    <li>python38Packages.setuptools-declarative-requirements</li>
    <li>python39Packages.img2pdf</li>
    <li>python39Packages.pikepdf</li>
    <li>python39Packages.pypiserver</li>
    <li>python39Packages.pytest-helpers-namespace</li>
    <li>python39Packages.setuptools-declarative-requirements</li>
  </ul>
</details>

**Pull request merged**